privacyIDEAADFSProvider/PIResponse.cs

```diff
@@ -68,17 +81,46 @@ public static PIResponse FromJSON(string json, PrivacyIDEA privacyIDEA)
                     ret.Message = (string)detail["message"];
                     ret.Type = (string)detail["type"];
 
-                    JToken multiChallenge = detail["multi_challenge"];
-                    if (multiChallenge!= null)
+                    JArray multiChallenge = detail["multi_challenge"] as JArray;
```

Comment:
See my comment about the MultiChallenge.

In a better world I would like it to be `detail["challenges"]` and thus maybe the plugin should already reflect this, since the term Multi_Challenge clashes with the repeated challenges.

(Open for discussion ;-)

Reply:
I dont know if i understand this correctly. I can not write `details["challenges"]` since that field does not exist in the response of the server yet. I could only rename the variable...

Reply:
My fault. THis is the originial JSON response you are parsing. So this is totally right.